### DISCUSSION TITLE
How to record individual credit card transactions?

### DISCUSSION THREAD

**Discussion by godlike64** [firefly-iii/firefly-iii#8589] [Firefly III] [answered]
So I started testing out Firefly-III. After an initial setup, I was able to import data directly from Nordigen (GoCardless) and have all the transactions of my main bank account for the last 90 days into Firefly. I then started fiddling around trying to organize everything, and here's my problem:

- Firefly documentation states that credit cards should be asset accounts. That's fine.
- In the transactions I imported, the monthly payments for my credit cards are present (which are paid in full). I can create an expense account to assign them, as a way of categorizing, so that they don't show up as going to "(unknown destination account)".

This creates two problems (the way I see it)
- My expense account that belongs to my credit card shows a positive balance, and I don't think that should be right? Maybe?
- While I can't access this data through Nordigen, I want to be more granular and also record the individual credit card purchases I make. So over the course of the month I would be registering all my purchases <somewhere> and then (via the import) when the payment is made, <somewhere>'s balance would be back to zero.

According to Firefly, that <somewhere> should be an asset account. However I can't register transactions with a negative value (so that they get balanced out with my monthly payment. Am I doing something wrong here? What would be the appropriate way of dealing with this in my situation?

**Comment by JC5** [comment]
Yes, expense accounts get a positive balance, that's normal. So no worries there.

The payment comes from an asset account, so it would be a transfer from your normal bank account to your credit card.

**Reply by godlike64** [reply]
I understand that, and the import took care of that. But if I wanted to be more granular about it and manually record every credit card purchase, how would I go about it so that it balances out with the payment from my main asset account at the end of the period?

To be clear, I don't have this information from my bank via GoCardless (this = every individual credit card purchase, amount, date, etc.). I cannot import it since my bank only lets me download a custom XLS file. Hence I want to add it manually.

Edit: to clarify what I'm looking for, as it stands right now I can know how much I spent on my credit card, total, every month. That does not give me information of the breakdown of each purchase (which one was clothing, which one was food, which one was an impulsive Amazon purchas, etc.). My aim with recording credit card purchases as they happen is to be able to track this and get a better sense of where my money is going.

**Reply by JC5** [reply]
I'm not sure what the question is. 

> how would I go about it so that it balances out with the payment from my main asset account at the end of the period?

Isn't this a question of tracking your credit card expenses, and creating withdrawals in Firefly III? That's what I do. My bank doesn't offer me downloads either, but I can track transactions through their app, so I copy them manually.

The only drawback is that the monthly cut-off point for credit card transactions is on the 25th, so expenses around that time are always maybe/maybe not part of the transfer.

**Reply by godlike64** [reply]
Yes, that is what I want to do but I don't know what to do exactly. I'm guessing you have your main asset account which is your bank account, another asset account for your credit card and an expense account for your credit card which is where your monthly credit card payment goes (from bank account main asset to credit card expense). But where do the individual purchases go? Ideally they should be involving your credit card expense account and your credit card asset account, but the amount in these transactions cannot be negative.

I guess what I'm asking is: what exactly do you do when you are recording your individual credit card purchases? Sorry if I'm being annoying with this, I'm just trying to understand the Firefly model and how to tailor my needs around it.

**Reply by JC5** [reply, marked as answer]
Ah yes, not a problem! Simply put, I have:

- One asset account: my main bank account
- One asset account: my credit card
- X expense accounts: all the shops.

1. So expenses from the credit card go from CC to ExpenseAccountX, Y and Z. The credit card drops into the negative, unless you set a virtual balance.
2. At the end of the month, a transer from asset 1 to asset 2.

Makes sense?

**Reply by godlike64** [reply]
OK I think I'm understanding it. Bear with me if you will:

1. Each of the CC purchases is registered between the CC asset account and the corresponding expense account. So you have one expense account for clothing, one for food, or what have you. Makes sense! I'm guessing these are normal withdrawals, hence the CC asset account falls into the negative.
2. At the end of the month, since you are transferring from asset 1 to asset 2, this would mean that I have to convert the imported transaction (which is a withdrawal) to a transfer, right? And I'm guessing even after doing this, if I import again, it will not be overwritten because there's some kind of internal ID still present that will tell Firefly "hey, this imported transaction is the same as this one that has been converted to a transfer".

**Reply by JC5** [reply]
1. Yes. Personally, I make expense accounts per store, but this is up to you.
2. Yes, exactly. Even after a conversion, the transaction will not be imported again. Depending on the GoCardless data, it may be possible to map them automatically so it is a transfer right away, and you will not need a conversion.

**Reply by godlike64** [reply]
Roger that. Yes, I was getting confused because I didn't know that I was supposed to convert those withdrawals into transfers, as withdrawals only allow me to specify an expense account (I thought I was supposed to create a CC expense account). Now it makes sense. Converting the imported withdrawals into a transfer from one asset account (bank account) to another (credit card) allows me to balance out the second asset account. It also now makes sense that expense accounts will always get a positive balance.

I think this is an important example that should be up in the documentation. It may not be the sole way of doing things, but it may help new users (maybe those without that much of fresh accounting knowledge - my last stint of getting involved in anything accounting was at high school xD) get to know how to model their data in Firefly.

I do have a very small gripe with the way expense accounts are shown on the main expense account page (e.g. "/accounts/show/#"). You get the graph which appears to start at 0 and increases over the period. However Y column does not start at 0, but at the accumulated amount at the beginning of the period. This is confusing, since the title for the page mentions the time period (e.g. "between February 1st, 2024 and February 29th, 2024 "). If I want to know historically how much I have spent on a certain expense account, I would choose a report. If I am looking at an overview of the account and presented with data for a specific period, I am expecting to see how much money I spent in that period. While the total amount is present lower on the page, the chart (showing the scale in total amount since the very first transaction registered) is not useful to get a sense of this data "at a glance". It's a personal gripe, but just thought I'd mention it.

Anyway, thanks!

**Reply by JC5** [reply]
I'll make sure to expand this page with some details, thanks!

https://docs.firefly-iii.org/explanation/financial-concepts/transactions/

**Comment by Kibukx** [comment]
Hey @JC5, I just started with Firefly-iii and ran into the same issue but not quite sure how to resolve this (FYI I'm in Mexico). In my case I use 2 of my credit cards for msi (meses sin intereses = interest free payment) where the shop/entity gives for example 3, 6, etc. months interest free payment.

example I made a purchase of 6,000 pesos which I can pay in 6 months no interests. 6,000 / 6 = 1000, so for the next 6 months i will be paying 1,000 on my X credit card. I want to be able to record the total amount (my debt in my CC) as well as visualize how much I'll be paying on my credit cards when I start making more purchases with or without interest free payment. This would help me setup a budget of some sort and make sure I only use my credit cards when I'm within that budget amount. Hope that makes sense. Thanks in advance!

**Reply by EduardoPazz** [reply]
🆙

**Comment by Jihad** [comment]
Thanks a lot @JC5 for an amazing app and your responses here. It helped me a lot even ones from 2017 :D.

One idea i had wanted to see if it makes sense, Would the Credit Card Statement “due amount” makes sense to be as a “bill”? If every 4th of the month i get a statement with “$1000” for example, that is due on the 25th. Usually I pay it in full on 25th, and it’s shown as credit in next month statement as reference of paying past dues.

Or maybe it wouldn’t have any effect just for organizing stuff?

**Reply by JC5** [reply]
What I do personally is add a piggy bank for the due amount. The expenses on the credit card are already part of my budgets, so they don't need another bill.

**Reply by Jihad** [reply]
@JC5 Interesting idea! Yes agree a "bill" doesn't make sense. I just Scraped 12 months of CC Statements PDFs and added more than thousand transaction. So those thoughts came while reconciling "to make sure everything accurate"; the duration of Statement and Due amount bill were important mental anchors to match things.

Useful Tip: Tagging CC Transaction with related statement Month helped as well (i.e. 'CC_NOV24')

Side note, Used the API heavily with N8N and Nocodb, and i got to say again, amazing work! The API is a godsend. I would really recommend adding it to N8N official library, so many things can be done & reviewed in bulk with little effort.

**Reply by Jihad** [reply]
@JC5 Quick update, Since i'm using n8n heavily, made a custom node for Firefly API; [n8n-nodes-fireflyiii](https://www.npmjs.com/package/n8n-nodes-fireflyiii), using it to automate couple things 😆.

**Reply by JC5** [reply]
Very cool, thanks!

**Reply by a-hersa** [reply]
I found this old post looking for how to import credit card individual payments to firefly. I may be in a similar case as others where GoCardless imports accounts only and we only see a single payment each month for credit card.

My bank allows me to download csv and pdf but that has to be manual.

How do you all go about that to see each individual payment in Firefly? Creating an assets account for credit card and import there each transaction is the first step. How do you automate the process as much as possible?

**Reply by JC5** [reply]
Mmm, since the import of CSV files is fairly manual that's tricky. However, you can run the data importer over the command line. As long as the CSV file is in place (for example, stored under the right name) it could be imported by a cron job.

**Reply by a-hersa** [reply]
Does GoCardless provides CC data for your bank? Manually downloading the csv and importing it is the only way I see it feasible...
